packages/features/ee/teams/components/MemberListItem.tsx

```diff
@@ -410,28 +716,76 @@ export default function MemberListItem(props: Props) {
           </DialogContent>
         </Dialog>
       )}
-
-      {showChangeMemberRoleModal && (
-        <MemberChangeRoleModal
-          isOpen={showChangeMemberRoleModal}
-          currentMember={props.team.membership.role}
-          teamId={props.team?.id}
-          memberId={props.member.id}
-          searchTerm={props.searchTerm}
-          initialRole={props.member.role as MembershipRole}
-          onExit={() => setShowChangeMemberRoleModal(false)}
+      {state.inviteMember.showModal && (
+        <MemberInvitationModalWithoutMembers
+          teamId={props.team.id}
+          token={props.team.inviteToken?.token}
+          hideInvitationModal={() => {
+            dispatch({
+              type: "CLOSE_MODAL",
+            });
+          }}
+          showMemberInvitationModal={true}
+          onSettingsOpen={() => {
+            dispatch({
+              type: "INVITE_MEMBER",
+              payload: {
+                showModal: false,
+              },
+            });
+            dispatch({
+              type: "INVITE_LINK_SETTING",
+              payload: {
+                showModal: true,
+              },
+            });
+          }}
+        />
+      )}
+      {state.inviteLinkSetting.showModal && props.team.inviteToken && (
+        <InviteLinkSettingsModal
+          isOpen={true}
+          teamId={props.team.id}
+          token={props.team.inviteToken.token}
+          expiresInDays={props.team.inviteToken.expiresInDays || undefined}
+          onExit={() => {
+            dispatch({
+              type: "INVITE_MEMBER",
+              payload: {
+                showModal: true,
+              },
+            });
+            dispatch({
+              type: "INVITE_LINK_SETTING",
+              payload: {
+                showModal: false,
+              },
+            });
+          }}
```

Comment:
why did you move this modal inside `MemberListItem`  ? we should keep the modal outside of `MemberListItem` as it doesn't depend upon any member

Reply:
Also, the memberinvitationmodel